proto/src/determined/experiment/v1/experiment.proto

```diff
@@ -139,3 +139,17 @@ message ValidationHistoryEntry {
   // the validation.
   float searcher_metric = 3;
 }
+
+// File node is one node of file in experiment model definition file tree.
+message FileNode {
+  // Path of file.
+  string path = 1;
+  // Modification time of file.
+  google.protobuf.Timestamp modified_time = 2;
+  // Number of bytes in file content.
+  int32 content_length = 3;
+  // Is this a directory.
+  bool is_dir = 4;
+  // MIME type of file.
+  string content_type = 5;
```

Comment:
we don't know the mime-type of files, so we probably shouldn't be pretending that we do.

Also, we support symbolic links in archives, but I don't see that those are actually supported in this message type.  How would we support those?  301 redirects or something?